src/rust/mangaworld/template/src/template.rs

```diff
@@ -9,9 +10,40 @@ use aidoku::{
 	Chapter, DeepLink, Filter, FilterType, Manga, MangaContentRating, MangaPageResult, MangaViewer,
 	Page,
 };
-
+use aidoku::std::html::Node;
 use crate::helper::{get_chapter_number, get_search_url, manga_status};
 
+fn get_html_with_cookie(url: String, referer: Option<&str>) -> Result<Node> {
+	let cookie_value = "MWCookie=";
+	let mut html = Request::new(&url, HttpMethod::Get)
+		.header("Cookie", cookie_value);
+
+	if let Some(ref_url) = referer {
+		html = html.header("referer", ref_url);
+	}
+
+	let html = html.html()?;
+	let raw_html = html.outer_html().to_string();
+
+	if raw_html.contains("MWCookie") {
+		if let Some(cookie_start) = raw_html.find("MWCookie=") {
+			let cookie_start_index = cookie_start + "MWCookie=".len();
+			if let Some(cookie_end) = raw_html[cookie_start_index..].find(";") {
+				let new_cookie_value = &raw_html[cookie_start_index..cookie_start_index + cookie_end];
+				let new_cookie = &*(cookie_value.to_string() + new_cookie_value);
+				let mut req = Request::new(&url, HttpMethod::Get)
+					.header("Cookie", new_cookie);
+				if let Some(ref_url) = referer {
+					req = req.header("referer", ref_url);
+				}
+				return req.html();
+			}
+		}
+	}
```

Comment:
```suggestion
	if let Some(cookie_start) = raw_html.find(cookie_value) {
		let cookie_start_index = cookie_start + cookie_value.len();
		if let Some(cookie_end) = raw_html[cookie_start_index..].find(";") {
			let new_cookie_value = &raw_html[cookie_start_index..cookie_start_index + cookie_end];
			let new_cookie = &*(cookie_value.to_string() + new_cookie_value);
			let mut req = Request::new(&url, HttpMethod::Get)
				.header("Cookie", new_cookie);
			if let Some(ref_url) = referer {
				req = req.header("referer", ref_url);
			}
			return req.html();
		}
	}
```
- you don't need to check if the html contains the text; that's what find does anyways
- use the variable you already defined instead of writing it multiple times